src/libsync/creds/oauth.cpp

```diff
@@ -102,10 +103,16 @@ class RegisterClientJob : public QObject
 private:
     void rgisterClientOnline()
     {
-        auto job = new OCC::SimpleNetworkJob(_account->sharedFromThis(), this);
-        job->setAuthenticationJob(true);
-        connect(job, &OCC::SimpleNetworkJob::finishedSignal, this, [this, job] {
-            const auto data = job->reply()->readAll();
+        QNetworkRequest req;
+        req.setUrl(_registrationEndpoint);
+        req.setAttribute(HttpCredentials::DontAddCredentialsAttribute, true);
+        const QJsonObject json({ { QStringLiteral("client_name"), QStringLiteral("%1 %2").arg(Theme::instance()->appNameGUI(), OCC::Version::versionWithBuildNumber().toString()) },
+            { QStringLiteral("redirect_uris"), QJsonArray { QStringLiteral("http://127.0.0.1") } },
+            { QStringLiteral("application_type"), QStringLiteral("native") },
+            { QStringLiteral("token_endpoint_auth_method"), QStringLiteral("client_secret_basic") } });
+        auto reply = _account->networkAccessManager()->post(req, QJsonDocument(json).toJson());
```

Comment:
I don't think we use the network job class for the core auth job.
One of the main issues I can currently see is that the we will lose the support for timeouts.
This change is kind of unrelated to the pr and I will move it out at some point.